src/anchor-click-interceptor.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  closestByTag,
+} from './dom';
+import {dev} from './log';
+import {urlReplacementsForDoc} from './url-replacements';
+
+/** @private @const {string} */
+const ORIG_HREF_ATTRIBUTE = 'data-a4a-orig-href';
+
+/**
+ * Registers a handler that performs URL replacement on the href
+ * of an ad click.
+ * @param {!./service/ampdoc-impl.AmpDoc} ampdoc
+ * @param {!Window} win
+ */
+export function installAnchorClickInterceptor(ampdoc, win) {
+  win.document.documentElement.addEventListener('click',
+      maybeExpandUrlParams.bind(null, ampdoc), /* capture */ true);
+}
+
+/**
+ * Handle click on links and replace variables in the click URL.
+ * The function changes the actual href value and stores the
+ * template in the ORIGINAL_HREF_ATTRIBUTE attribute
+ * @param {!./service/ampdoc-impl.AmpDoc} ampdoc
+ * @param {!Event} e
+ */
+function maybeExpandUrlParams(ampdoc, e) {
+  const target = closestByTag(dev().assertElement(e.target), 'A');
+  if (!target || !target.href) {
```

Comment:
logically it's a bit different. it could be the case that it does not have href but have ORIG_HREF_ATTRIBUTE